packages/opentelemetry-core/src/utils/environment.ts

```diff
@@ -64,6 +64,28 @@ function parseNumber(
   }
 }
 
+/**
+ * Parses a LogLevelString as a LogLevel
+ * @param key
+ * @param environment
+ * @param values
+ */
+function parseLogLevel(
+  key: keyof ENVIRONMENT,
+  environment: ENVIRONMENT_MAP | ENVIRONMENT,
+  values: ENVIRONMENT_MAP
+) {
+  const value = values[key];
+  const logLevelStrings = Object.keys(LogLevel).filter(
+    k => isNaN(Number(k)) === true
+  ) as LogLevelString[];
+  const regex = new RegExp(logLevelStrings.join('|'), 'i');
```

Comment:
Dynamically built regexes are rarely a good idea in my experience.

Reply:
any anecdote in particular to share?

Reply:
Not in particular